File: pset6/readability/readability.py
from cs50 import get_string

# Get user input
text = get_string("Text: ")

# Initialisa variables
letters = 0
words = 1
sentences = 0

# Iterate over characters in text and count elements
for c in text:
    if c.isalpha():
        letters += 1
    if c == " ":
        words += 1
    if c in ['.', '!', '?']:
        sentences += 1

# Coleman-Liau formmula
L = 100 * letters / words
S = 100 * sentences / words

index = round(0.0588 * L - 0.296 * S - 15.8)

# Output
if index >= 16:
    print("Grade 16+")
elif index <= 0:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Declare variables
    int c = 0;
    float letters = 0;
    float words = 1;
    float sentences = 0;
    string text = get_string("Text: ");

    // Iterate over characters in text
    while (text[c] != '\0')
    {
        if (text[c] == ' ')
        {
            if (text[c + 1] != ' ')
            {
                words++;
            }
        }

        // Check whether element is in alphabet
        if (((65 <= (int) text[c] && (int) text[c] <= 90)) || ((97 <= (int) text[c]) && (int) text[c] <= 122))
        {
            letters++;
        }

        // Check for '?', '.' or '!'
        if (text[c] == '.' || text[c] == '?' || text[c] == '!')
        {
            sentences++;
        }
        c++;
    }
    // Calculations
    float L = 100 * letters / words;
    float S = 100 * sentences / words;
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    // Final conditions
    if (index >= 16)
    {
        printf("\nGrade 16+\n");
    }
    else if (index < 1)
    {
        printf("\nBefore Grade 1\n");
    }
    else
    {
        printf("\nGrade %d\n", (int) index);
    }
}